### DISCUSSION TITLE
How to perform unit test and integration test on RISCV architecture

### DISCUSSION THREAD

**Discussion by StrayLittlePunk** [rust-embedded/rust-raspberrypi-OS-tutorials#104] [Tutorial Questions]
Thank you very much to the author for writing such an excellent article, but I have a question. How does `cargo test` entrust `qemu` to run the test, I imitated this method to run the test on the RISCV kernel, but the test results cannot be printed

I read [Controlling Test Kernel Execution](https://github.com/rust-embedded/rust-raspberrypi-OS-tutorials/tree/master/13_integrated_testing#controlling-test-kernel-execution) and be confused, Can someone tell me how to perform test on RISCV, Please ?

**Comment by andre-richter** [comment]
Hi,

if you exchange the QEMU binary `qemu-system-aarch64` for the corresponding RISCV one, the tests should load correctly.
However, changing to RISCV would mean a lot of other changes as well.

You would have a different BSP, with different peripherals (different UART) and all that. Did you make all those other changes as well?
